Project finished

Timowest12 opened the pull request:
In this project @Munsa1 and I created a recipe app.
This app is capable of:

- Adding recipes that will be added to the database
- Add foods
- Add pre-added foods to a recipe
- Make a shopping list based on the foods in the recipes
- update and remove recipes/foods

our design uses bootstrap, and the login system is done with devise and cancancan

uwadonat commented on app/views/recipe_foods/new.html.erb:
```diff
@@ -0,0 +1,22 @@
+    placeholder: '1', min: 1%>
+  </div>
+</div>
+
+<div class="mb-3"><%= form.submit "Add", class:"btn btn-dark" %></div>
+<%end%>
+<div class="mb-3">
```
- [ ] Currently `add button` is not working
![Screenshot from 2022-03-25 16-23-17](https://user-images.githubusercontent.com/65860678/160140847-1d08e77e-d127-4107-aec2-6a957e9f43ee.png)

Timowest12 replied:
@uwadonat this button seems to work fine for both of us, I've also tested on multiple devices and browsers and they all work fine, are you sure it doesn't work on your end?

Munsa1 replied:
You are adding negative values, which is not allowed on purpose as you can not prepare food in a negative timeline. Using a positive value works.

uwadonat replied:
Ok. If this is the case. I think it should be better to add some validations. No need to have negative numbers as option.

Thanks.

ahmetbozaci replied:
### OPTIONAL
- [ ] Nice to add validation for a negative value. But when ı try to add negative value ı am not get any message. It will be better if show the message to the user.

uwadonat commented on app/views/devise/sessions/new.html.erb:
```diff
@@ -0,0 +1,24 @@
+<h4>Log in</h4>
+
+<%= form_for(resource, as: resource_name, url: session_path(resource_name)) do
+|f| %>
+<div class="mb-3">
+  <%= f.label :email, class:'form-label' %><br />
+  <%= f.email_field :email, autofocus: true, autocomplete: "email",
+  class:'form-control' %>
+</div>
+
+<div class="mb-3">
+  <%= f.label :password, class:'form-label' %><br />
+  <%= f.password_field :password, autocomplete: "current-password",
+  class:'form-control' %>
+</div>
```
- [ ] Kindly fix your Homepage and add more styling for all pages
![Screenshot from 2022-03-25 16-21-10](https://user-images.githubusercontent.com/65860678/160141912-c09b8a80-b475-47f7-a89c-6f32ed07894f.png)
.

Munsa1 replied:
This issue has been resolved

ahmetbozaci replied:
- Well done for adding style. The design looks better.

uwadonat reviewed (changes requested):
Hi **[Munsa1](https://github.com/Munsa1),** 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!


## Highlights
- Your commit messages are very descriptive ✔️
- No linter errors  ✔️

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

Timowest12 commented:
Added styling to:

Prevent flash messages from rendering out of view
Give more margin to the button
Changed styling of the navbar to give the site a nice colour scheme
Added a nice font

ahmetbozaci commented on app/views/foods/new.html.erb:
```diff
@@ -0,0 +1,20 @@
+<h4>Add Food</h4>
+<%= form_with model: @food do |form| %>
+<div class="mb-3">
+  <%= form.label :name, class:'form-label' %><br />
+  <%= form.text_field :name, class:'form-control' %>
+</div>
+
+<div class="mb-3">
+  <%= form.label :measurement_unit, class:'form-label' %><br />
+  <%= form.text_field :measurement_unit, class:'form-control' %>
+</div>
+
+<div class="mb-3">
+  <%= form.label :price, class:'form-label' %><br />
+  <%= form.number_field :price, class:'form-control' %>
+</div>
+
+<div class="actions mb-5"><%= form.submit "Add", class:'btn btn-dark' %></div>
+
+<% end %>
```
### OPTIONAL
- [ ] Your food model has some validations. But when ı try to add food that is against validations ı am not getting any message. If users get any message about why food is not added it will be better.
**You can show error messages for other models if they have validations.**

ahmetbozaci commented on app/views/devise/sessions/new.html.erb:
```diff
@@ -0,0 +1,24 @@
+<h4>Log in</h4>
+
+<%= form_for(resource, as: resource_name, url: session_path(resource_name)) do
+|f| %>
+<div class="mb-3">
+  <%= f.label :email, class:'form-label' %><br />
+  <%= f.email_field :email, autofocus: true, autocomplete: "email",
+  class:'form-control' %>
+</div>
+
+<div class="mb-3">
+  <%= f.label :password, class:'form-label' %><br />
+  <%= f.password_field :password, autocomplete: "current-password",
+  class:'form-control' %>
+</div>
```
### OPTIONAL
- [ ] When ı try login unexist user ı am not getting any messages. It will be better if show messages to users.

ahmetbozaci reviewed (approved):
# STATUS: APPROVED 🥳 🥳 🥳 
![](https://media.giphy.com/media/3pZipqyo1sqHDfJGtz/giphy.gif)

@Munsa1 @Timowest12  ✋🏼 
### Your project is great. 🎉 I approve of it. There is nothing else to say other than... it's time to merge it. You can submit this project on your dashboard.

___ 

### Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._
@Timowest12 
- Your commit messages should always start with capital letters and they should be an imperative tense. I see some of them are not like that. Instead of writing `Added style or Created the file` you should write `Add style or Create the file`
- ***I write it as optional. But this is a [general requirement](https://github.com/microverseinc/curriculum-transversal-skills/blob/main/documentation/articles/professional_repo_rules.md#:~:text=The%20commit%20message%20should%20be%20written%20in%20the%20imperative%20tense%20and%20capitalized) for every project. So you should pay attention to this in your next projects.***

**OPTIONAL**

- You can change them by using `git-rebase`. If you want to know how to change ı leave a few resources.
- [Video - How to change commit messages](https://www.loom.com/share/00771f369b7b47fc968c019ecf0d3ba6)
- [Article - How to create good commit messages](https://www.freecodecamp.org/news/level-you-up-to-awesome-commit-messages-a85558cb90e8/)
___

### Cheers and Happy coding! 💻 👏

![aproved](https://media.giphy.com/media/d31w24psGYeekCZy/giphy.gif)

___

_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form.](https://airtable.com/shrQAqnBwek5a0O0s)_


_If anything seems to be unclear to you, reach out and ask anything by commenting on this Pull Request and mentioning me as [ahmetbozaci](https://github.com/ahmetbozaci), I will receive a notification and will respond._
___
[GitHub](https://github.com/ahmetbozaci) | [LinkedIn](https://linkedin.com/in/ahmetbozaci) | [Twitter](https://twitter.com/ahmtbozaci)

_Reviewed by [Ahmet Bozacı ](https://github.com/ahmetbozaci)_
